go/worker/storage/committee/node.go

```diff
@@ -1066,10 +1063,27 @@ func (n *Node) worker() { // nolint: gocyclo
 				if !syncing.outstanding.contains(rootType) && syncing.awaitingRetry.contains(rootType) {
 					syncing.scheduleDiff(rootType)
 					fetcherGroup.Add(1)
+					select {
+					case <-time.After(backoff.timeout()):
```

Comment:
Actually, the backoff having effect in the next call to `triggerRoundFetches` is desired/ok. Since the problematic case that causes hogging the CPU is actually the repeated calling of `triggerRoundFetches` not that the loop here is too large.
But it does maybe make sense to move the "waiting for timeout"  at the start of the function to make this clearer.

---

But this comment actually made me look into what is causing this in the first place, and there might be even a simpler fix, that the clickup task didn't consider. Let me open a separate PR where the authors of the code can comment on it if they deem it correct.

Reply:
Opened also https://github.com/oasisprotocol/oasis-core/pull/5342

@kostko which one seems preferable?

Reply:
I think #5342 is better.